### PR TITLE
Pagination UI

### PR DESCRIPTION
**Pull request by Megha-Dev-19**
image

- when no data exists
image

@robert-zaremba let me know if you need any changes to the UI

### CODE REVIEW

**Review by rmeena840 — COMMENTED** [review]
Backedn PR has to be merged first.

**Comment by robert-zaremba** [comment]
UI looks good.  I didn't check the code though

File: src/pages/AllDao.vue [hunk 1]
```diff
@@ -233,54 +301,104 @@ export default defineComponent({
...
+							token_id: +item.assetId,
 							token_name: tokenData.name as string,
 							name: globalState.get(GLOBAL_STATE_MAP_KEY.DaoName) as string,
 							link: globalState.get(GLOBAL_STATE_MAP_KEY.Url) as string,
-						});
+						};
+						this.dataSource.push(data);
+
+						// pushing data to store only if it doesn't exists
+						let dataExists: boolean;
```
Comment (rmeena840) [inline src/pages/AllDao.vue]: I would suggest this.
```suggestion
						let isCached: boolean;
```

**Review by rmeena840 — CHANGES_REQUESTED** [review]
Reviewed. Creating an enum for navigation would save a lot of code lines. I would suggest going with this approach.

File: src/pages/AllDao.vue [hunk 2]
```diff
@@ -233,54 +301,104 @@ export default defineComponent({
...
+						if (this.formState.psqlData) {
+							dataExists = this.formState.psqlData.has(+item.appId);
+						} else {
+							this.formState.psqlData = new Map();
+							dataExists = false;
```
Comment (rmeena840) [inline src/pages/AllDao.vue]: No need to set `false ` here. initialize line 359 'isCached' with false.

**Review by rmeena840 — CHANGES_REQUESTED** [review]

File: src/pages/AllDao.vue [hunk 3]
```diff
@@ -107,30 +107,21 @@
...
 					v-if="type === 'prev'"
-					@click="
-						fetchDataForDAO(
-							null,
-							null,
-							ROWS_PER_PAGE,
-							currentPageCursor.startCursor
-						)
-					"
+					@click="handlePaginationCall(PaginationCallType.PREVIOUS)"
```
Comment (rmeena840) [inline src/pages/AllDao.vue]:
```suggestion
					@click="handlePaginationCall(PaginationCallType.NAV_PREV)"
```

File: src/pages/AllDao.vue [hunk 4]
```diff
@@ -107,30 +107,21 @@
...
 					v-else-if="type === 'next'"
-					@click="
-						fetchDataForDAO(
-							ROWS_PER_PAGE,
-							currentPageCursor.endCursor,
-							null,
-							null
-						)
-					"
+					@click="handlePaginationCall(PaginationCallType.NEXT)"
```
Comment (rmeena840) [inline src/pages/AllDao.vue]:
```suggestion
					@click="handlePaginationCall(PaginationCallType.NAV_NEXT)"
```

File: src/pages/AllDao.vue [hunk 5]
```diff
@@ -230,20 +221,42 @@ export default defineComponent({
...
 				startCursor: null,
 			},
+			PaginationCallType,
 		};
 	},
 	methods: {
+		handlePaginationCall(type: PaginationCallType, pageNumber?: string) {
+			switch (type) {
+				case PaginationCallType.PREVIOUS:
+					this.fetchDataForDAO(
```
Comment (rmeena840) [inline src/pages/AllDao.vue]:
```suggestion
					this.fetchDaoData(
```

**Review by rmeena840 — APPROVED** [review]
Pre-approving. Look at suggestions.

File: src/pages/AllDao.vue (hunk 3 above)
Comment (Megha-Dev-19) [reply src/pages/AllDao.vue]: updated

File: src/pages/AllDao.vue (hunk 5 above)
Comment (Megha-Dev-19) [reply src/pages/AllDao.vue]: updated

**Pull request merged**